text/trace/0189-instrumentation-layers-suppression.md

```diff
@@ -0,0 +1,222 @@
+- Trace SDK: Add span creation option to set `InstrumentationType`
+  - During span creation, checks if span should be suppressed (if there is another one for kind + type on the parent `Context`) and if there is, returns a *suppressed* span, which is
+    - non-recording
+    - propagating (carries parent context)
+    - does not become current (i.e. `makeCurrent` call with it is no-op)
+- OTel SDK SHOULD allow suppression strategy configuration
+  - suppress nested by kind (e.g. only one CLIENT allowed)
+  - suppress nested by kind + convention it follows (only one HTTP CLIENT allowed, but outer DB -> nested HTTP is ok)
+  - suppress none
+
+Note: some conventions may explicitly include others (e.g. [FaaS](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/faas.md) may include [HTTP](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/http.md)). For the purpose of this document, we assume span follows single convention and instrumentation MUST include explicitly mentioned sub-conventions (except general) only.
+[General](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/span-general.md) convention attributes are allowed on all spans when applicable, pure generic instrumentation are not suppressed.
+
+#### Suppression Example
+
+- Instrumentation - HTTP Client 1:
+  - starts span
+    - SDK checks if there is a span of the same kind + type on the parent context
+    - there is no similar span
+    - SDK returns new span
+  - stores it in the context (presumably current)
+    - SDK stores span in context behind `ContextKey` which is combination of kind and type (depending on configuration)
+- Instrumentation - HTTP Client 2 (in scope of span1):
+  - [Optional suggestion]: add API to check if span should be started (optimization for power-users)
+  - starts span
+    - SDK checks if there is a span of the same kind + type on the parent context
+    - there is one already
+    - SDK returns suppressed (non-recording, propagating span, it's also never becomes current).
+  - HTTP client continues instrumentation with suppressed span
+    - adds attributes: noop
+    - injects context: duplicate work, since parent context is already injected and suppressed span carries the same context
+    - makes suppressed span current: current is not modified, scope is noop
+- Instrumentation - TLS (imaginary, in scope of span2)
+  - works as if it's in scope of span1
+  - current is span1
+  - if explicit parent was used (span2), it carries span1 context
+
+Suppression logic is configurable and encapsulated in span builder SDK implementation - specific library instrumentation should not depend on configuration, e.g.:

```

Comment:
libraries never think about the configuration. I believe suppression by kind + type is a good default since it shows all layers suppressing duplicate instrumentations only. 
I don't have too strong an opinion - vendors/users should be able to easily change it.